components/capture-build-scan-url-maven-extension/src/main/java/com/gradle/CaptureBuildScansListener.java

```diff
@@ -74,14 +80,26 @@ private static String urlEncode(String str) {
 
     private void capturePublishedBuildScan(BuildScanApi buildScan) {
         buildScan.buildScanPublished(scan -> {
+            // abort with an error message written to the error file if no GE server URL is available
+            if (buildScan.getServer() == null || buildScan.getServer() .equals("https://unknown.server.com")) {
```

Comment:
From what I can tell in my local testing, it's much to late to check this here as a Build Scan will not be published if the GE URL is not set.